### PR TITLE
ROADMAP: Add section on "Automated compatibility testing"

### PR DESCRIPTION
**Pull request by mxinden**
I would like to propose (a) including _automated compatibility testing_ into our roadmap and (b) prioritizing it within the _Evolve_ section.

What do folks think?

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -243,6 +244,21 @@ ping-pong).
 
 - [Protocol Select specification](https://github.com/libp2p/specs/pull/349)
 
+### 🧪 Automated compatibility testing
+
+**What?** There are more than 6 implementations of the [libp2p specification] in
+different languages. We need to ensure compatibility of all combination of these
```
Comment (marten-seemann) [inline ROADMAP.md:250]:
```suggestion
different languages. We need to ensure compatibility of all combinations of these
```

File: ROADMAP.md [hunk 2]
```diff
@@ -243,6 +244,21 @@ ping-pong).
...
+implementations. Given the number of libp2p implementations and the amount of
+libp2p protocols, verifying compatibility on a continuous basis can only be
+tackled in an automated fashion.
+
+**Why?** Automated compatibility testing allows us to move fast with confidence
+and will increase trust in the libp2p project as a whole.
```
Comment (marten-seemann) [inline ROADMAP.md:256]: And move some tests currently at the IPFS layer to where they actually belong.

**Review by marten-seemann — APPROVED** [review]

File: ROADMAP.md (hunk 2 above)
Comment (mxinden) [reply ROADMAP.md:256]: Agreed that we should do that, though I would consider it a detail. Do you feel strongly on mentioning it in our roadmap?

Comment (marten-seemann) [reply ROADMAP.md:256]: No, that was just a general comment.

**Pull request merged**